uhubctl.c

```diff
@@ -1051,9 +1058,12 @@ int main(int argc, char *argv[])
                 for (port=1; port <= hubs[i].nports; port++) {
                     if ((1 << (port-1)) & ports) {
                         int port_status = get_port_status(devh, port);
+                        printf("Port Status: %i\n", port_status);
+                        if (opt_action == POWER_TOGGLE)
+                            request = (port_status == 0) ? LIBUSB_REQUEST_SET_FEATURE : LIBUSB_REQUEST_CLEAR_FEATURE;
```

Comment:
`port_status == 0` does not necessarily mean power is off.
You need to use `port_status & power_mask` instead

Reply:
Ok, just tested it with port_status and it worked. I'll change …